test/viewmodels/CashAccountDailyStatementsViewModelSpec.scala

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels
+
+import models.{CashDailyStatement, CashTransactions, Declaration, Payment, Transaction, Transfer, Withdrawal}
+import org.scalatest.Assertion
+import play.api.Application
+import play.api.i18n.Messages
+import utils.SpecBase
+import utils.TestData.*
+import helpers.Formatters
+import utils.Utils.{LinkComponentValues, h2Component, linkComponent}
+
+import java.time.LocalDate
+
+class CashAccountDailyStatementsViewModelSpec extends SpecBase {
+
+  "apply" should {
+
+    "return correct list of dailyStatements" when {
+
+      "cash transactions are available" in new Setup {
+        val dailyStatementsViewModel: CashAccountDailyStatementsViewModel =
+          CashAccountDailyStatementsViewModel(cashTransactions)
+
+        dailyStatementsViewModel.dailyStatements.size must be > 0
+        shouldContainCorrectContentsForDailyStatements(dailyStatementsViewModel.dailyStatements)
+        shouldContainTransactionForLastSixMonthsHeading(dailyStatementsViewModel)
+        dailyStatementsViewModel.hasTransactions mustBe true
+      }
+
+      "cash transactions are not present" in new Setup {
+        val cashAccountDailyStatementsViewModelWithNoTransactions: CashAccountDailyStatementsViewModel =
+          CashAccountDailyStatementsViewModel(
+            cashTransactions.copy(pendingTransactions = Seq(), cashDailyStatements = Seq()))
+
+        cashAccountDailyStatementsViewModelWithNoTransactions.dailyStatements mustBe empty
+        cashAccountDailyStatementsViewModelWithNoTransactions.hasTransactions mustBe false
+        shouldContainTransactionForLastSixMonthsHeading(cashAccountDailyStatementsViewModelWithNoTransactions)
+      }
+    }
+
+  }
+
+  private def shouldContainCorrectContentsForDailyStatements(actualStatements: Seq[DailyStatementViewModel])
+                                                            (implicit msgs: Messages): Assertion = {
+
+    val expectedDailyStatements: Seq[DailyStatementViewModel] = populateDailyStatViewModelFromDailyCashTransactions()
+
+    //TODO To be uncommented once pending statements are added in daily statements
```

Comment:
is this comment genuinely todo or can it be deleted?

Reply:
Yes it is. This will be uncommented once design team will come back regarding pending statements queries